lfs/client.go

```diff
@@ -235,9 +246,13 @@ func Batch(objects []*objectResource, operation string) ([]*objectResource, erro
 		}
 
 		if IsAuthError(err) {
-			Config.SetAccess("basic")
-			tracerx.Printf("api: batch not authorized, submitting with auth")
-			return Batch(objects, operation)
+			if isNtlmRequest(res) {
+				toggleAuthType("ntlm", "api: response indicates ntlm, submitting with %s auth")
+				return Batch(objects, operation)
+			} else {
+				toggleAuthType("basic", "api: batch not authorized, submitting with %s auth")
```

Comment:
I think it makes sense for `toggleAuthType()` to write a standard message, instead of requiring a custom one for each auth type. I think all auth types can just use a format string like "api: response indicates %q authentication. Resubmitting"

Also, I don't see a reason for `isNtlmRequest()`. I think I'd prefer something like `responseAuthType(res)` that read "www-authenticate" or "lfs-authenticate".